config/opal_check_pmi.m4

```diff
@@ -303,6 +303,19 @@ AC_DEFUN([OPAL_CHECK_PMIX],[
 
            # if it does exist, then we need to parse it to find
            # the actual release series
+           AS_IF([test "$opal_external_pmix_version_found" = "0"],
+                 [AC_MSG_CHECKING([version 4x])
+                  AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+                                                      #include <pmix_version.h>
+                                                      #if (PMIX_VERSION_MAJOR < 4L)
```

Comment:
Is it better to write `PMIX_VERSION_MAJOR != 4L` so we don't need to update this for PMIx 5?

Reply:
Actually, we chose this method so it won't have to be changed - we are checking that the version is 4 or higher. If it is, then all the code that uses PMIx 4 or lower APIs will still be good. If you someday update to PMIx 5 APIs, then you would need to check for 5 or higher.

Reply:
Ok, as long as you guys promise to never break backward compatibility.  :neckbeard: 🤞

Reply:
Yeah, that is the promise we made!

Reply:
We'll be just fine as long as nobody retires!  😉